src/data/mv_polynomial/cardinal.lean

```diff
@@ -89,14 +89,35 @@ calc #(mv_polynomial_fun σ R) = #R + #σ + #(ulift bool) :
 
 namespace mv_polynomial
 
+lemma cardinal_lift_mk_eq_max {σ : Type u} {R : Type v}
+  [nonempty σ] [comm_semiring R] [nontrivial R] :
+  #(mv_polynomial σ R) = max (max (cardinal.lift.{u} $ #R) $ cardinal.lift.{v} $ #σ) ℵ₀ :=
+begin
+  haveI : infinite (σ →₀ ℕ) := infinite_iff.2 ((le_max_right _ _).trans (mk_finsupp_nat σ).symm.le),
+  refine (mk_finsupp_lift_of_infinite _ R).trans _,
+  rw [mk_finsupp_nat, max_assoc, lift_max, lift_aleph_0, max_comm],
+end
+
+lemma cardinal_mk_eq_max {σ R : Type u} [nonempty σ] [comm_semiring R] [nontrivial R] :
+  #(mv_polynomial σ R) = max (max (#R) (#σ)) ℵ₀ :=
+by rw [cardinal_lift_mk_eq_max, lift_id, lift_id]
+
+lemma cardinal_lift_mk_le_max {σ : Type u} {R : Type v} [comm_semiring R] :
+  #(mv_polynomial σ R) ≤ max (max (cardinal.lift.{u} $ #R) $ cardinal.lift.{v} $ #σ) ℵ₀ :=
+begin
+  casesI subsingleton_or_nontrivial R,
+  { exact (le_one_iff_subsingleton.2 finsupp.coe_fn_injective.subsingleton).trans
+      (le_max_of_le_right one_le_aleph_0) },
+  casesI is_empty_or_nonempty σ,
+  { rw [((is_empty_ring_equiv R σ).to_equiv.trans equiv.ulift.{u}.symm).cardinal_eq, mk_ulift],
```

Comment:
You could separate this `cardinal_eq` bit into its own `simp` lemma. Can't hurt.